src/main/groovy/me/tatarka/RetrolambdaPluginAndroid.groovy

```diff
@@ -123,7 +123,8 @@ public class RetrolambdaPluginAndroid implements Plugin<Project> {
         }
 
         JavaCompile compileUnitTest = (JavaCompile) project.tasks.find { task -> 
-            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") 
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") ||
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJavaWithJavac")
```

Comment:
That is strange then. I know I changed it from equals to startsWith in the past because of this issue with the task being renamed. The change seemed to work fine when I was testing it. Mind doing a little bit more investigating on why it's not working or providing a sample project?